src/modm/architecture/interface/i2c_multiplexer.hpp

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2018, Sascha Schade
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_I2C_MULTIPLEXER_HPP
+#define MODM_I2C_MULTIPLEXER_HPP
+
+#include <modm/processing/resumable.hpp>
+
+namespace modm
+{
+
+/**
+ * Interface of an I2C Multiplexer.
+ *
+ * All I2C multiplexer driver should implement this interface.
+ *
```

Comment:
Please put all classes in this file into the proper doxygen group, `modm_i2c_multiplexer` I believe. You can check the output via `(cd modm/docs && doxygen doxyfile)`.

Reply:
Like so?

Reply:
Uh, sorry, the generator in `:docs` uses module naming, so it would be `modm_architecture_i2c_multiplexer`.